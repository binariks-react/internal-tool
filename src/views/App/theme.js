export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
