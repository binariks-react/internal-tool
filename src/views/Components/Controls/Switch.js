import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const getColorBackGround = (props) => {
  if (props.disable) {
    if (props.checked) {
      return props.theme.colors.border;
    }
    return props.theme.colors.white;
  }
  if (props.checked) {
    return props.theme.colors.success;
  }
  return props.theme.colors.white;
};

const getColorSwitcher = (props) => {
  if (props.disable) {
    if (props.checked) {
      return props.theme.colors.text;
    }
    return props.theme.colors.border;
  }
  if (props.checked) {
    return props.theme.colors.white;
  }
  return props.theme.colors.border;
};


const getColorBorder = (props) => {
  if (props.disable) {
    return props.theme.colors.border;
  }
  if (props.checked) {
    return props.theme.colors.success;
  }
  return props.theme.colors.border;
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  cursor: ${props => props.disable ? 'not-allowed' : 'pointer'};
`;

const SwitchWrapper = styled.div`
  background-color: ${getColorBackGround};
  border-radius: 100px;
  border: .0625rem solid ${getColorBorder} ;
  width: 2.1875rem;
  height: 1.125rem;
  position: relative;
  display: flex;
  align-items: center;
  margin-right: 5px;
  transition: 0.3s;
`;

const Switcher = styled.div`
  content: '';
  position: absolute;
  left: ${props => props.checked ? 1.25 : 0.1875}rem;
  width: .75rem;
  height: .75rem;
  background-color: ${getColorSwitcher};
  border-radius: 6.25rem;
  transition: 0.2s ease-in-out;
`;

const Label = styled.label`
  cursor: ${props => props.disable ? 'not-allowed' : 'pointer'};
  line-height: 15px;
  color: ${props => props.theme.colors.text};
  font-size: 15px;
`;


const Switch = ({ preventDefault, stopPropagation, ...props }) => (
  <Wrapper
    {...props} onClick={
      (e) => {
        if (preventDefault) e.preventDefault();
        if (stopPropagation) e.stopPropagation();
        if (!props.disable) props.onClick(!props.checked);
      }}
  >
    <SwitchWrapper {...props}>
      <Switcher {...props} />
    </SwitchWrapper>
    {!!props.title && <Label {...props}>{props.title}</Label>}
  </Wrapper>
);


const propTypes = {
  checked: PropTypes.bool,
  onClick: PropTypes.func,
  title: PropTypes.string,
  preventDefault: PropTypes.bool,
  stopPropagation: PropTypes.bool,
  disable: PropTypes.bool,
};

const defaultProps = {
  checked: false,
  onClick: () => {
  },
  title: '',
  preventDefault: true,
  stopPropagation: true,
  disable: false,
};

Switch.propTypes = propTypes;
Switch.defaultProps = defaultProps;

export default Switch;
