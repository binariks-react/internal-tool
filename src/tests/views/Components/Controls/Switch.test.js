import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';
import theme from 'views/App/theme';
import 'jest-styled-components';
import Switch from '../../../../views/Components/Controls/Switch';

describe('Switch', () => {
  const preventDefault = jest.fn();
  const stopPropagation = jest.fn();
  beforeEach(() => {
    preventDefault.mockClear();
    stopPropagation.mockClear();
  });
  describe('snapshot', () => {
    it('onClick', () => {
      const onClick = jest.fn();
      const button = mount(
        <ThemeProvider theme={theme}>
          <Switch onClick={onClick} />
        </ThemeProvider>
      );

      button.find(Switch).simulate('click', { preventDefault, stopPropagation });
      expect(onClick).toHaveBeenCalledTimes(1);
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(stopPropagation).toHaveBeenCalledTimes(1);
    });

    it('onClick - defaultProps', () => {
      expect(Switch.defaultProps.onClick).toBeDefined();
      expect(Switch.defaultProps.onClick()).toBeUndefined();
    });

    it('onClick - preventDefault', () => {
      const onClick = jest.fn();
      const button = mount(
        <ThemeProvider theme={theme}>
          <Switch preventDefault={false} onClick={onClick} />
        </ThemeProvider>
      );

      button.find(Switch).simulate('click', { preventDefault, stopPropagation });
      expect(onClick).toHaveBeenCalledTimes(1);
      expect(preventDefault).toHaveBeenCalledTimes(0);
      expect(stopPropagation).toHaveBeenCalledTimes(1);
    });

    it('onClick - stopPropagation', () => {
      const onClick = jest.fn();
      const button = mount(
        <ThemeProvider theme={theme}>
          <Switch stopPropagation={false} onClick={onClick} />
        </ThemeProvider>
      );

      button.find(Switch).simulate('click', { preventDefault, stopPropagation });
      expect(onClick).toHaveBeenCalledTimes(1);
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(stopPropagation).toHaveBeenCalledTimes(0);
    });
  });
});
