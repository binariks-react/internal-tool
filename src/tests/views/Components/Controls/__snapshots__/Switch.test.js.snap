// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:active > div > div {
  width: 1rem;
  left: 0.1875rem;
}

.c0:active > div {
  box-shadow: none;
}

.c1 {
  background-color: #ffffff;
  border-radius: 100px;
  border: .0625rem solid #e1e5eb;
  width: 2.1875rem;
  height: 1.125rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 5px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  outline: none;
  box-shadow: 0 3px 8px 1px rgba(0,0,0,.2);
}

.c2 {
  content: '';
  position: absolute;
  left: 0.1875rem;
  width: .75rem;
  height: .75rem;
  background-color: #e1e5eb;
  border-radius: 6.25rem;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
</div>
`;
